app/apps/manifest.rb

```diff
@@ -49,36 +50,128 @@ def self.load_from_string(yaml)
+  # @option opts [String] :icon The icon used on the dashboard, optionally a Font Awesome tag
+  # @option opts [String] :role Dashboard categorization
+  # @option opts [String] :url An optional redirect URL
   def initialize(opts)
     raise InvalidContentError.new unless(opts && opts.is_a?(Hash))
 
-    # merge with defaults, ignoring nil
-    opts = defaults.merge(opts) { |key, oldval, newval| newval || oldval  }
+    @manifest_options = opts.with_indifferent_access
+  end
+
+  # The name of the application
+  #
+  # @return [String] name as string
+  def name
+    @manifest_options[:name] || ""
+  end
+
+  # The description of the application
+  #
+  # @return [String] description as string
+  def description
+    @manifest_options[:description] || ""
+  end
+
+  # The icon used on the dashboard, optionally a Font Awesome tag
+  #
+  # @return [String] icon as string
+  def icon
+    @manifest_options[:icon] || ""
+  end
+
+  # Return the optional redirect URL string
+  #
+  # @return [String] url as string
+  def url
+    @manifest_options[:url] || ""
+  end
+
+  # Return the app category
+  #
+  # @return [String] category as string
+  def category
+    @manifest_options[:category] || ""
+  end
 
-    @name = opts.fetch("name")
-    @description = opts.fetch("description")
-    @category = opts.fetch("category")
-    @subcategory = opts.fetch("subcategory")
-    @icon = opts.fetch("icon")
-    @role = opts.fetch("role")
-    @url = opts.fetch("url")
+  # Return the app subcategory
+  #
+  # @return [String] subcategory as string
+  def subcategory
+    @manifest_options[:subcategory] || ""
   end
 
+  # Return the app role
+  #
+  # @return [String] role as string
+  def role
+    @manifest_options[:role] || ""
+  end
+
+  # Manifest objects are valid
+  #
+  # @return [true] Always return true
   def valid?
     true
   end
 
+  # Manifest objects exist
+  #
+  # @return [true] Always return true
   def exist?
     true
   end
+
+  # Save the current manifest to a path.
+  #
+  # @param [String, Pathname] path The full path of the file to be saved as string or Pathname object
+  #
+  # @return [true, false] true if the file is saved successfully
+  def save(path)
+
+    File.open(path.to_s, 'w') do |f|
+      f.write(self.to_yaml)
+    end
+
+    true
+  rescue
+    # TODO Add a custom exception here to track why it erred. IO? Permissions? etc.
+    false
+  end
+
+  # Merge the contents of a hash into this Manifest's options
+  #
+  # @param [Hash, Manifest] opts The options to update
+  #
+  # @return [Manifest] A new manifest with the updated options
+  def merge(opts)
+    raise InvalidContentError.new unless(opts && (opts.is_a?(Hash) || opts.is_a?(Manifest)))
+
+    Manifest.new(@manifest_options.merge(opts.to_h))
+  end
+
+  # Creates a hash of the object's current state.
+  #
+  # @return [Hash] A hash representation of the Manifest object.
+  def to_h
+    @manifest_options
```

Comment:
Why don't we call `#compact` here? And instead wait until the `def to_yaml` portion.